lib/nat/interface.go

```diff
@@ -19,11 +19,19 @@ const (
 	UDP Protocol = "UDP"
 )
 
+type IPVersion int8
+
+const (
+	DualStack = iota
+	IPv4Only
+	IPv6Only
+)
```

Comment:
When "IP" came out big, most developers didn't think of a future when there would be more than the current IP (v4) at the time.  Let's not repeat that wrong assumption now, by making a `DualStack` enumeration that might end up being a `TripleStack` sometime.

How about `IPvAny`? Because a socket will either use a specific address, or bind to all available addresses regardless of address family.

Reply:
Looking at the adoption rate of IPv6, we can safely skip future proofing.

Reply:
Probably, but I already wrote the commit. The $2^{69}$ possible `::/56` subnets with GUAs in `2000::/3` should last for a while though.